contrib/dracut/90zfs/mount-zfs.sh.in

```diff
@@ -57,30 +57,15 @@ ZFS_POOL="${ZFS_DATASET%%/*}"
 
 if import_pool "${ZFS_POOL}" ; then
 	# Load keys if we can or if we need to
-	if [ $(zpool list -H -o feature@encryption $(echo "${ZFS_POOL}" | awk -F\/ '{print $1}')) == 'active' ]; then
+	if [ $(zpool list -H -o feature@encryption $(echo "${ZFS_POOL}" | awk -F\/ '{print $1}')) = 'active' ]; then
 		# if the root dataset has encryption enabled
-		if $(zfs list -H -o encryption "${ZFS_DATASET}" | grep -q -v off); then
-			# figure out where the root dataset has its key, the keylocation should not be none
-			while true; do
-				if [[ $(zfs list -H -o keylocation "${ZFS_DATASET}") == 'none' ]]; then
-					ZFS_DATASET=$(echo -n "${ZFS_DATASET}" | awk 'BEGIN{FS=OFS="/"}{NF--; print}')
-					if [[ "${ZFS_DATASET}" == '' ]]; then
-						rootok=0
-						break
-					fi
-				else
-					rootok=1
-					break
-				fi
-			done
-			[[ "${rootok}" -eq 0 ]]&& return 1
+		ENCRYPTIONROOT=$(zfs get -H -o value encryptionroot ${ZFS_DATASET})
```

Comment:
I think we do actually want to encapsulate this one because you can have a space in the filesystem name >_>